src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -586,15 +586,51 @@ public override void VisitFunctionCallSyntax(FunctionCallSyntax syntax)
 
                 switch (bindings[syntax])
                 {
-                    case ErrorSymbol errorSymbol:
+                    case UnassignableSymbol errorSymbol:
                         // function bind failure - pass the error along
-                        return new ErrorTypeSymbol(errors.Concat(errorSymbol.GetDiagnostics()));
+                        return new UnassignableTypeSymbol(errors.Concat(errorSymbol.GetDiagnostics()));
 
                     case FunctionSymbol function:
-                        return GetFunctionSymbolType(syntax, function, syntax.Arguments, argumentTypes, errors);
+                        return GetFunctionSymbolType(function, syntax.OpenParen, syntax.CloseParen, syntax.Arguments, argumentTypes, errors);
 
                     default:
-                        return new ErrorTypeSymbol(errors.Append(DiagnosticBuilder.ForPosition(syntax.Name.Span).SymbolicNameIsNotAFunction(syntax.Name.IdentifierName)));
+                        return new UnassignableTypeSymbol(errors.Append(DiagnosticBuilder.ForPosition(syntax.Name.Span).SymbolicNameIsNotAFunction(syntax.Name.IdentifierName)));
+                }
+            });
+
+        public override void VisitInstanceFunctionCallSyntax(InstanceFunctionCallSyntax syntax)
+            => AssignType(syntax, () => {
+
+                base.VisitInstanceFunctionCallSyntax(syntax);
+
+                var errors = new List<ErrorDiagnostic>();
+                
+                var baseType = typeManager.GetTypeInfo(syntax.BaseExpression);
+                CollectErrors(errors, baseType);
+
+                if (errors.Any())
+                {
+                    return new UnassignableTypeSymbol(errors);
```

Comment:
Do we actually need to short-circuit the type check here? Typically, we only short-circuit to avoid errors caused by other errors (or if the logic requries non-error type info to work right). Can you try removing to see what happens?

Reply:
Removing this condition has the following result:

Example: a.b.c.bar().baz

Removing the condition results in a red squiggly on "a" ("a" does not exist in current context) and "bar" ("bar" does not exist in current context), b and c do not have a red squiggly.

Adding the condition back results in a red squiggly on "a" same message as above.

I think we need to short circuit so we can stop the validation as soon as we see the first error, otherwise we might end up with multiple errors on the same line. Thoughts?